### ISSUE TITLE
Resizing PVC

### ISSUE THREAD

**Issue by niklaskorz** [hetznercloud/csi-driver#8]
Changing the size of a PersistentVolumeClaim after it has already been created and applying the manifest currently results in the following error:
`forbidden: only dynamically provisioned pvc can be resized and the storageclass that provisions the pvc must support resize`

Are there any plans to support resizing persistent volumes created by the Hetzner CSI driver in the future? If not, is there any way to do this manually?

**Comment by thcyron** [comment]
The [CSI spec](https://github.com/container-storage-interface/spec/blob/master/spec.md) does not support resizing volumes yet. There are [efforts](https://github.com/kubernetes/enhancements/issues/556) to implement that, but we’re not there yet.

**Comment by dedene** [comment]
@thcyron I believe https://github.com/container-storage-interface/spec/pull/334 adds the necessary interface for implementing resizing volumes to the CSI spec. But the support for this in the volume plugin in the core kubernetes code is still work in progress.

**Comment by manvalls** [comment]
there's [a sidecar container](https://github.com/kubernetes-csi/external-resizer) now, also:
https://github.com/kubernetes/enhancements/issues/556
seems the actual feature will be landing on 1.14?

**Comment by deacix** [comment]
What is the state?

**Comment by thcyron** [comment]
There’s no estimate right now when we are going to implement that feature.

**Comment by niklaskorz** [comment]
Does resizing the associated Hetzner volume have any effect, or would the change just not be recognized by Kubernetes?

**Comment by thcyron** [comment]
> Does resizing the associated Hetzner volume have any effect, or would the change just not be recognized by Kubernetes?

Since the underlying filesystem won‘t be resized to fit the new volume size, the change won’t be recognized.

**Comment by vitobotta** [comment]
Hi, is this planned yet? Thanks

**Comment by codeagencybe** [comment]
Any updates on this please?

**Comment by thcyron** [comment]
We don’t have any plans right now to implement resizing, but since this seems to be a highly requested feature, I’ll check if we are able to tackle this in the next few weeks.

**Comment by smoulderme** [comment]
@thcyron @LKaemmerling Any updates on this issue?

**Comment by LKaemmerling** [comment]
We are still working on this. I hope that we have a solution (release) within the next two weeks.

**Comment by driehle** [comment]
Resizing the volumen in Hetzner Console, going via SSH to the worker where the volume is currently attachted and running `resize2fs /dev/disk/by-id/scsi-0HC_Volume_xxxx` works. The volume is then resized, `df -ah` both on the host as well as inside the pods show the correct values. However, Kubernets and Rancher still show the old size.

**Comment by LKaemmerling** [comment]
We have officially released v1.2.0 of the driver with support for resizing:
https://github.com/hetznercloud/csi-driver/blob/master/CHANGES.md#v120

**Comment by manvalls** [comment]
Is this enough for upgrading in k8s 1.16?
```
kubectl apply -f https://raw.githubusercontent.com/hetznercloud/csi-driver/v1.2.0/deploy/kubernetes/hcloud-csi.yml
```
or do I need to run this before
```
kubectl delete -f https://raw.githubusercontent.com/hetznercloud/csi-driver/v1.1.5/deploy/kubernetes/hcloud-csi.yml
```

**Comment by LKaemmerling** [comment]
Running the apply for v1.2.0 should be enough.
```
kubectl apply -f https://raw.githubusercontent.com/hetznercloud/csi-driver/v1.2.0/deploy/kubernetes/hcloud-csi.yml
```

**Comment by driehle** [comment]
Ich bekomme nach dem Upgrade auf 1.2.0 wie oben vorgeschlagen folgende Fehlermeldung beim Anlegen eines neuen PVC:

failed to provision volume with StorageClass "hcloud-volumes": error generating accessibility requirements: error getting CSINode for selected node "worker3": csinodes.storage.k8s.io "worker3" is forbidden: User "system:serviceaccount:kube-system:hcloud-csi" cannot get resource "csinodes" in API group "storage.k8s.io" at the cluster scope

**Comment by driehle** [comment]
Already reported with #77 and most likely fixed with #78.
